src/mmtk.rs

```diff
@@ -188,9 +188,18 @@ impl<VM: VMBinding> MMTK<VM> {
         }
 
         // TODO: This probably does not work if we have multiple MMTk instances.
-        VM_MAP.boot();
         // This needs to be called after we create Plan. It needs to use HeapMeta, which is gradually built when we create spaces.
-        VM_MAP.finalize_static_space_map(heap.get_discontig_start(), heap.get_discontig_end());
+        VM_MAP.finalize_static_space_map(
+            heap.get_discontig_start(),
+            heap.get_discontig_end(),
+            &mut |start_address| {
```

Comment:
The use of callback here is not quite intuitive. Some comments are needed. When does this closure get called? What is the argument? What does the closure do?

Reply:
I added comments to `finalize_static_space_map` and this closure.